### PR TITLE
[WIP] Docs

### PR DESCRIPTION
**Pull request by andrehp**

### CODE REVIEW

**Comment by felipejfc** [comment]
❤️

File: docs/API.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+
+### Handler example
+
+Below is a very barebones example of a handler definition, for a complete working example, check the [cluster demo](https://github.com/topfreegames/pitaya/tree/master/examples/demo/cluster).
+
+```go
+import (
+	"github.com/topfreegames/pitaya"
+  "github.com/topfreegames/pitaya/component"
```
Comment (cscatolini) [inline docs/API.md]: fix ident

File: docs/API.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+The clients can call the handler by calling `serverType.handlerName.methodName`.
+
+
+### Routing messages
+
+Messages are forwarded by pitaya to the appropriate server type, and custom routers can be added to the application by calling `pitaya.AddRoute`, it expects two arguments:
+
+* `serverType`: the server type of the target requests to be routed
+* `routingFunction`: the routing function with the signature `func(*session.Session, *route.Route, []byte, map[string]*cluster.Server) (*cluster.Server, error)`, it receives the user's session, the route being requested, the message and the map of valid servers of the given type.
```
Comment (cscatolini) [inline docs/API.md]: Please clarify how should be the map of valid servers. Is the key a server id or type?

**Comment by andrehp** [comment]
@cscatolini I added part of the architecture docs, what do you think about the format?

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (cscatolini) [inline docs/Makefile]: missing new line

File: session/session.go [hunk 4]
```diff
@@ -74,10 +74,10 @@ type HandshakeData struct {
 	User map[string]interface{} `json:"user,omitempty"`
 }
 
-// Session represents a client session which could storage temp data during low-level
-// keep connected, all data will be released when the low-level connection was broken.
-// Session instance related to the client will be passed to Handler method as the first
-// parameter.
+// Session represents a client session, which can store data during the connection.
+// All data is be released when the low-level connection is broken.
```
Comment (cscatolini) [inline session/session.go]: /is be released/data is released/

File: docs/architecture.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+#### Backend sessions
+
+Backend sessions have access to the sessions through the handler's methods, but they have some limitations and special characteristics. Changes to session variables must be pushed to the frontend server by calling `s.PushToFront` (this is not needed for `s.Bind` operations), setting callbacks to session lifecycle operations is also not allowed. One can also not retrieve a session by user ID from a backend server.
+
+### Service discovery
+
+Servers operating in cluster mode must have a service discovery client to be able to work, Pitaya comes with a default client using etcd, which is used if no other client is defined. The service discovery client is responsible for registering the server and keeping the list of valid server updated, as well as providing information about requested servers as needed.
```
Comment (cscatolini) [inline docs/architecture.md]: /of valid server/of valid servers/

Comment (cscatolini) [reply docs/architecture.md]: /to work, Pitaya/to work. Pitaya/

**Comment by andrehp** [comment]
@cscatolini I merged the docs with master, but I will keep working on them, I will create a new PR as soon as I have something new.

Comments on everything so far are welcome

**Pull request merged**